src/openms/include/OpenMS/FORMAT/HANDLERS/MzMLHandler.h

```diff
@@ -1492,6 +1533,7 @@ namespace OpenMS
         if (spectrum_data_.size() >= options_.getMaxDataPoolSize())
         {
           populateSpectraWithData();
+          spectrum_data_.reserve(options_.getMaxDataPoolSize());
```

Comment:
hmm good point: basically after it once reached capacity, doing `reserve` is not helping any more. So I moved the `reserve` up to `setOptions` which is the one place where we change how much space we will need at the very beginning